connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -0,0 +1,215 @@
+                Builder builder = new Builder(credentialsProvider);
+                builder.connection = connection;
+                return builder;
+            }
+
+            /**
+             * Factory method for creating a new Location Configuration builder.
+             *
+             * @param connectionId A Connection id that the {@link ConnectionApiClient} can use to fetch the
+             * associated Connection object.
+             * @return The Builder object itself for chaining.
+             */
+            public static Builder withConnectionId(String connectionId, LocationCredentialsProvider credentialsProvider) {
+                Builder builder = new Builder(credentialsProvider);
+                builder.connectionId = connectionId;
+                return builder;
+            }
+
+            private Builder(LocationCredentialsProvider credentialsProvider) {
+                this.credentialsProvider = credentialsProvider;
+            }
+
+            /**
+             * @param connectionApiClient an optional {@link ConnectionApiClient} that will be used for the ConnectButton
+             * instead of the default one.
+             * @return The Builder object itself for chaining.
+             */
+            public Builder setConnectionApiClient(ConnectionApiClient connectionApiClient) {
+                this.connectionApiClient = connectionApiClient;
+                return this;
+            }
+
+            /**
+             * @param inviteCode an optional string as the invite code, this is needed if your service is not yet
+             * published on IFTTT Platform.
+             * @return The Builder object itself for chaining.
+             * @see ConnectionApiClient.Builder#setInviteCode(String)
+             */
+            public Builder setInviteCode(String inviteCode) {
+                this.inviteCode = inviteCode;
+                return this;
+            }
+
+            public LocationConfiguration build() {
+                if (connection == null && connectionId == null) {
+                    throw new IllegalStateException("Either connection or connectionId must be non-null.");
+                }
+
+                LocationConfiguration configuration =
+                        new LocationConfiguration(credentialsProvider, connectionApiClient);
+                configuration.connection = connection;
+                configuration.connectionId = connectionId;
+                configuration.inviteCode = inviteCode;
+                return configuration;
+            }
+
+        }
+
+        private LocationConfiguration(LocationCredentialsProvider credentialsProvider, @Nullable ConnectionApiClient connectionApiClient) {
+            this.credentialsProvider = credentialsProvider;
+            this.connectionApiClient = connectionApiClient;
+        }
+    }
+
+    private static final class UserTokenAsyncTask extends android.os.AsyncTask<Void, Void, String> {
```

Comment:
Similar to the above comment, since we already have something very similar in connect-button, we should consider reusing.